compiler/rustc_macros/src/diagnostics/fluent.rs

```diff
@@ -189,31 +189,49 @@ pub(crate) fn fluent_messages(input: proc_macro::TokenStream) -> proc_macro::Tok
             if let Entry::Message(Message { id: Identifier { name }, attributes, .. }) = entry {
                 let _ = previous_defns.entry(name.to_string()).or_insert(ident_span);
 
-                // `typeck-foo-bar` => `foo_bar` (in `typeck.ftl`)
-                // `const-eval-baz` => `baz` (in `const_eval.ftl`)
-                let snake_name = Ident::new(
+                if name.contains('-') {
```

Comment:
Could you submit a PR to the dev guide documenting this and updating the examples there too? Please ping me on it :)

(this can be after this lands, just so that it happens)